pkg/scheduler/predicates/stable_scheduling.go

```diff
@@ -102,11 +99,9 @@ func (p *stableScheduling) Filter(instanceName string, pod *apiv1.Pod, nodes []a
 				return []apiv1.Node{node}, nil
 			}
 		}
-		msg := fmt.Sprintf("cannot run on its previous node %q", nodeName)
-		p.recorder.Event(pod, apiv1.EventTypeWarning, UnableToRunOnPreviousNodeReason, msg)
-	} else {
-		glog.V(2).Infof("no previous node exists for pod %q in TiDB cluster %s/%q", podName, ns, tcName)
+		return nodes, fmt.Errorf("cannot run %s/%s on its previous node %q", ns, podName, nodeName)
 	}
+	glog.V(2).Infof("no previous node exists for pod %q in TiDB cluster %s/%q", podName, ns, tcName)
```

Comment:
From the log message, there is no node for scheduling, but the following returns the `nodes` and the error is `nil`.

Reply:
this happens when we can't find the previous node of this TiDB member. e.g. the tidb pod is new, so we can run it on any node

Reply:
Yes, I got it. But we should use warning level log and also emit this event to users.

Reply:
After the change in this PR, the error returned here will be propagated to users.  @weekface can you help change the log level to warn and return the log message as an error too?

Reply:
comments addressed, @tennix @cofyc PTAL